src/main/scala/com/wavesplatform/state2/diffs/BlockDiffer.scala

```diff
@@ -17,32 +17,32 @@ object BlockDiffer extends ScorexLogging {
 
   val emptyDiff: Diff = Monoid[Diff].empty
   val rightEmptyDiff: Either[ValidationError, Diff] = Right(emptyDiff)
+  def rightDiff(diff: Diff): Either[ValidationError, Diff] = Right(diff)
 
   def apply(settings: FunctionalitySettings)(s: StateReader, block: Block): Either[ValidationError, BlockDiff] = {
 
     val txDiffer = TransactionDiffer(settings, block.timestamp, s.height + 1) _
 
-    val txsDiffEi = block.transactionData.foldLeft(rightEmptyDiff) { case (ei, tx) => ei match {
-      case Left(error) => Left(error)
-      case Right(diff) =>
-        txDiffer(new CompositeStateReader(s, diff.asBlockDiff), tx)
-          .map(newDiff => diff.combine(newDiff))
-    }
-    }
-
-    lazy val accountPortfolioFeesMap: List[(Account, Portfolio)] = block.feesDistribution.toList.map {
+    val accountPortfolioFeesMap: List[(Account, Portfolio)] = block.feesDistribution.toList.map {
       case (AssetAcc(account, maybeAssetId), feeVolume) =>
         account -> (maybeAssetId match {
           case None => Portfolio(feeVolume, LeaseInfo.empty, Map.empty)
           case Some(assetId) => Portfolio(0L, LeaseInfo.empty, Map(EqByteArray(assetId) -> feeVolume))
         })
     }
-    lazy val feeDiff = Monoid[Diff].combineAll(accountPortfolioFeesMap.map { case (acc, p) =>
+    val feeDiff = Monoid[Diff].combineAll(accountPortfolioFeesMap.map { case (acc, p) =>
       Diff(portfolios = Map(acc -> p))
     })
 
+    val txsDiffEi = block.transactionData.foldLeft(rightDiff(feeDiff)) { case (ei, tx) => ei match {
+      case Left(error) => Left(error)
+      case Right(diff) =>
+        txDiffer(new CompositeStateReader(s, diff.asBlockDiff), tx)
+          .map(newDiff => diff.combine(newDiff))
+      }
```

Comment:
what we actually need here is `foldM`, because we don't want to traverse all the collection after first error, yet we do it.